server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -304,7 +304,10 @@ public Snapshot revertSnapshot(Long snapshotId) {
 
         SnapshotInfo snapshotInfo = snapshotFactory.getSnapshot(snapshotId, dataStoreRole);
```

Comment:
@slavkap the datastore role for the snapshot is picked based to where it is stored (Image | Primary), doesn't matter what the config "snapshot.backup.to.secondary" is. So, no need to re-check _snapshotInfo_ with the "DataStoreRole.Primary" role.

https://github.com/apache/cloudstack/blob/e5a12ff12cd047c70b3c541c85512374741930fc/server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java#L1292-L1293

Reply:
@sureshanaparti, but there are cases where the method you shared returns `DataStoreRole.Image`, and the snapshot does not exist on secondary storage.

Reply:
@slavkap that method have to be fixed to return the correct datastore role, based on the snapshot location (primary or secondary storage).